### PR TITLE
feat: env-configurable retrieval stubs + wire similarity handler

### PR DESCRIPTION
**Pull request by jonahgabriel**
## Summary
- **OMN-6572**: Replace hardcoded `use_stub_handlers=True` with `OMNIMEMORY_USE_STUB_HANDLERS` env var; wire `ModelHandlerQdrantConfig` from `QDRANT_HOST`/`QDRANT_PORT`/`LLM_EMBEDDING_URL`
- **OMN-6573**: Add `HandlerSimilarityCompute` to `_HANDLER_SPECS` in `wiring.py`
- **OMN-6584**: Add unit tests for env-var-driven retrieval config

## Test plan
- [x] `uv run pytest tests/unit/test_dispatch_retrieval_config.py -v` passes (4/4)
- [x] `uv run mypy src/omnimemory --strict` passes (via pre-push hook)
- [ ] CI passes

## Summary by CodeRabbit

* **New Features**
  * Memory retrieval backend selection is now configurable via environment variables, allowing runtime switching between stub and production backends.

* **Infrastructure**
  * Extended handler verification to include an additional memory-related handler, improving service registration coverage.

* **Tests**
  * Added and updated unit tests covering environment-driven retrieval configuration, parsing robustness, defaults, and the expanded handler registration.

### CODE REVIEW

File: src/omnimemory/runtime/dispatch_handlers.py [hunk 1]
```diff
@@ -420,13 +421,33 @@ def create_memory_retrieval_dispatch_handler(  # stub-ok: references stub_handle
...
 
     _retrieval_handler: HandlerMemoryRetrieval | None = None
 
     async def _get_retrieval_handler() -> HandlerMemoryRetrieval:
         nonlocal _retrieval_handler
         if _retrieval_handler is None:
-            config = ModelHandlerMemoryRetrievalConfig(use_stub_handlers=True)
+            use_stubs = (
+                os.getenv("OMNIMEMORY_USE_STUB_HANDLERS", "true").lower() != "false"
+            )
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply src/omnimemory/runtime/dispatch_handlers.py]: Fixed in 3cdf099. Added `.strip()` before `.lower()` on the env var read. Also added `test_whitespace_padded_false_disables_stubs` to cover this edge case explicitly.

File: tests/unit/test_dispatch_retrieval_config.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        """OMNIMEMORY_USE_STUB_HANDLERS=false is correctly parsed."""
+        with patch.dict(os.environ, {"OMNIMEMORY_USE_STUB_HANDLERS": "false"}):
+            val = os.getenv("OMNIMEMORY_USE_STUB_HANDLERS", "true").lower() != "false"
+            assert val is False
+
+    def test_env_var_true_parsed(self) -> None:
+        """OMNIMEMORY_USE_STUB_HANDLERS=true keeps stubs."""
+        with patch.dict(os.environ, {"OMNIMEMORY_USE_STUB_HANDLERS": "true"}):
+            val = os.getenv("OMNIMEMORY_USE_STUB_HANDLERS", "true").lower() != "false"
+            assert val is True
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply tests/unit/test_dispatch_retrieval_config.py]: Fixed in 3cdf099. Extracted `build_retrieval_config_from_env()` as a module-level function from the dispatch handler closure. All tests now call this function directly, which exercises the same code path used at runtime (env var reading, config construction, Qdrant config wiring). Added 5 tests covering: default stubs, explicit false, explicit true, whitespace-padded false, and default Qdrant values when env vars are unset.
